Modules/_decimal/_decimal.c

```diff
@@ -3233,19 +3288,39 @@ dec_format(PyObject *dec, PyObject *args)
     }
 
     if (PyUnicode_Check(fmtarg)) {
-        fmt = (char *)PyUnicode_AsUTF8AndSize(fmtarg, &size);
+        fmt = PyUnicode_AsUTF8AndSize(fmtarg, &size);
         if (fmt == NULL) {
             return NULL;
         }
+        /* NOTE: If https://github.com/python/cpython/pull/29438 lands, the
+         *   format string manipulation below can be eliminated by enhancing
+         *   the forked mpd_parse_fmt_str(). */
         if (size > 0 && fmt[0] == '\0') {
             /* NUL fill character: must be replaced with a valid UTF-8 char
                before calling mpd_parse_fmt_str(). */
             replace_fillchar = 1;
-            fmt = dec_strdup(fmt, size);
-            if (fmt == NULL) {
+            fmt = fmt_copy = dec_strdup(fmt, size);
+            if (fmt_copy == NULL) {
                 return NULL;
             }
-            fmt[0] = '_';
+            fmt_copy[0] = '_';
+        }
+        /* Strip 'z' option, which isn't understood by mpd_parse_fmt_str().
+         * NOTE: fmt is always null terminated by PyUnicode_AsUTF8AndSize() */
+        char const *z_position = format_spec_z_search(fmt, size);
```

Comment:
> It looks as though this is still too lenient: format(Decimal("-0.000123"), ".2fz") (for example) produces 0.00 instead of raising ValueError.

Thank you-- I also realized that multi-byte fill character wasn't handled properly.  These should be resolved.